src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -33,25 +33,17 @@ public InstanceController(WebScriptHostManager scriptHostManager, ScriptSettings
         [HttpPost]
         [Route("admin/instance/assign")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public IActionResult Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
+        public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
         {
             var containerKey = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerEncryptionKey);
             var assignmentContext = encryptedAssignmentContext.Decrypt(containerKey);
-            return _instanceManager.StartAssignment(assignmentContext)
+            var result = await _instanceManager.Assign(assignmentContext);
+
+            return result
                 ? Accepted()
                 : StatusCode(StatusCodes.Status409Conflict, "Instance already assigned");
         }
 
-        [HttpGet]
-        [Route("admin/instance/status")]
-        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> GetInstanceStatus([FromQuery] int timeout = int.MaxValue)
```

Comment:
I don't think this is needed anymore?

Reply:
yeah. this is not needed anymore if we can get the status of admin/instance/assign from its return value.

Reply:
Yes, with a small fix. Previously we were swallowing exceptions from specialization and returning true, which could cause an incorrect 202 response. I'm now allowing exceptions to propagate, so if it fails you'll get a 500.